fe/fe-core/src/main/java/org/apache/doris/nereids/properties/DistributionSpecHash.java

```diff
@@ -174,7 +182,13 @@ public boolean satisfy(DistributionSpec required) {
             return containsSatisfy(requiredHash.getOrderedShuffledColumns());
         }
 
-        if (requiredHash.shuffleType == ShuffleType.NATURAL && this.shuffleType != ShuffleType.NATURAL) {
+        // when this shuffle type is natural, we allow contains satisfied for possible colocate-join
+        if (this.shuffleType == ShuffleType.NATURAL) {
```

Comment:
how can we handle left output's type is BUCKETED with possible bucket shuffle join?
If we need equalsSatisfy for BUCKETED type. the left hand may has been add Distribute on it, and has no chance to change to bucket join, am i right?